### PR TITLE
Bed 3789 - ADCSESC1

### PR DESCRIPTION
**Pull request by rvazarkar**
## Description
This MR completes the initial POC for ADCS ESC1, and includes the new shortcutting logic that will be used throughout the upcoming edges

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/analysis/post_integration_test.go [hunk 1]
```diff
@@ -82,3 +83,28 @@ func TestClearOrphanedNodes(t *testing.T) {
...
+	testContext := integration.NewGraphTestContext(t)
+	testContext.DatabaseTestWithSetup(func(harness *integration.HarnessDetails) {
+		harness.ShortcutHarness.Setup(testContext)
+	}, func(harness integration.HarnessDetails, db graph.Database) error {
+		firstSet := []*graph.Node{testContext.Harness.ShortcutHarness.Group1}
+		secondSet := []*graph.Node{testContext.Harness.ShortcutHarness.Group2}
+		groupExpansions, err := ad2.ExpandAllRDPLocalGroups(context.Background(), db)
+		require.Nil(t, err)
+		results := ad2.CalculateCrossProductNodeSets(firstSet, secondSet, groupExpansions)
+		//db.ReadTransaction(context.Background(), func(tx graph.Transaction) error {
```
Comment (JonasBK) [inline cmd/api/src/analysis/post_integration_test.go:97]: Do we want to save this for debugging or should we delete it?

File: packages/go/analysis/ad/adcs.go [hunk 2]
```diff
@@ -37,6 +37,152 @@ var (
 	EkuCertRequestAgent = "1.3.6.1.4.1.311.20.2.1"
 )
 
+func buildEsc1Cache(ctx context.Context, db graph.Database, enterpriseCAs, certTemplates []*graph.Node) (map[graph.ID]graph.NodeSet, error) {
+	cache := map[graph.ID]graph.NodeSet{}
+
+	return cache, db.ReadTransaction(ctx, func(tx graph.Transaction) error {
+		for _, ct := range certTemplates {
+			if firstDegreePrincipals, err := fetchFirstDegreeNodes(tx, ct, ad.Enroll, ad.GenericAll, ad.AllExtendedRights, ad.Owns); err != nil {
```
Comment (JonasBK) [inline packages/go/analysis/ad/adcs.go:45]: We should remove Owns

**Review by JonasBK — COMMENTED** [review]
Beast PR. Added two comments

**Review by JonasBK — APPROVED** [review]

**Comment by codydbentley** [comment]
Hey buddddyy, can we get a description on here please 🤣

File: cmd/api/src/test/integration/harnesses.go [hunk 3]
```diff
@@ -1082,6 +1082,204 @@ func (s *SearchHarness) Setup(graphTestContext *GraphTestContext) {
 	graphTestContext.UpdateNode(s.GroupLocalGroup)
 }
 
+type ADCSESC1Harness struct {
```
Comment (urangel) [inline cmd/api/src/test/integration/harnesses.go:1085]: Please include the harness svg and json representation(s) under cmd/api/src/test/integration/harnesses for reference

**Pull request merged**